src/plugins/cylc-object.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import CylcObjectPlugin from '@/components/cylc/cylcObject/plugin'
+import Vue from 'vue'
+
+// load Cylc Object mutations plug-in
+Vue.use(CylcObjectPlugin)
```

Comment:
This looks a bit redundant, but that's how we are loading plug-ins in Cylc UI… inherited this from Tim Creative's theme. It basically has `main.js` importing `src/plugins/index.js`.

And `src/plugin/index.js` is responsible for loading the plug-ins. Each plug-in file will import any objects it needs, and call `Vue.use($pluginObject)`. I've just kept the pattern here.